src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -187,20 +187,29 @@ private static void UpdateMetrics(ExecutionOptions opts, UInt256 effectiveGasPri
         {
             if (opts is ExecutionOptions.Commit or ExecutionOptions.None)
             {
-                float gasPrice = (float)((double)effectiveGasPrice / 1_000_000_000.0);
-                Metrics.MinGasPrice = Math.Min(gasPrice, Metrics.MinGasPrice);
-                Metrics.MaxGasPrice = Math.Max(gasPrice, Metrics.MaxGasPrice);
+                try
+                {
+                    float gasPrice = (float)((double)effectiveGasPrice / 1_000_000_000.0);
+                    if (float.IsInfinity(gasPrice))
+                    {
+                        return;
+                    }
```

Comment:
how can it be infinity?

Reply:
Can we just keep them as double then?

Reply:
Cap effectiveGasPrice at `float.MaxValue`?

Is `3 x 10^38` or `300_000_000_000_000_000_000 Ether` which is way higher than max supply; so isn't providing additional info at higher values